Tests/Recurly/Pager_Test.php

```diff
@@ -0,0 +1,103 @@
+    parent::_loadFrom($uri, $params);
+  }
+}
+Recurly_Resource::$class_map['mocks'] = 'Mock_Pager';
+
+class Mock_Item extends Recurly_Resource {
+  protected function getNodeName()            { return 'mock'; }
+  protected function getWriteableAttributes() { return array(); }
+  protected function getRequiredAttributes()  { return array(); }
+}
+Recurly_Resource::$class_map['mock'] = 'Mock_Item';
+
+
+class Recurly_PagerTest extends Recurly_TestCase
+{
+  function defaultResponses() {
+    return array(
+      array('GET', '/mocks', 'pager/index-1-200.xml'),
+      array('GET', 'http://example.com/mocks?cursor=1', 'pager/index-1-200.xml'),
+      array('GET', 'http://example.com/mocks?cursor=2', 'pager/index-2-200.xml'),
+      array('GET', 'http://example.com/mocks?cursor=3', 'pager/index-3-200.xml'),
+    );
+  }
+
+  private function assertIteratesCorrectly($pager, $count) {
+    // Initialization and enumeration
+    $pager->rewind();
+    $this->assertTrue($pager->valid());
+    $this->assertEquals(0, $pager->key());
+    $first = $pager->current();
+    $this->assertInstanceOf('Mock_Item', $first, 'Item 0 has correct type');
+    $this->assertEquals(1, $first->value, 'Item 0 has correct value');
+
+    for ($i = 1; $i < $count; $i++) {
+      $pager->next();
+      $this->assertTrue($pager->valid(), "Iterartor should still be valid");
+      // FIXME: This is the pattern keys follow for paged results, which creates
+      // duplicate values. I consider that a bug.
+      //   $this->assertEquals((1 - $i % $page_size) + 1, $pager->key());
+      // The nested results follow a much sainer:
+      //   $this->assertEquals($i, $pager->key());
+      $item = $pager->current();
+      $this->assertInstanceOf('Mock_Item', $item, "Item $i has correct type");
+      $this->assertEquals($i + 1, $item->value, "Item $i has correct value");
+    }
+
+    // End of the list
+    $pager->next();
+    $this->assertFalse($pager->valid(), "Iterator should now be invalid");
+
+    // Resets correctly
+    $pager->rewind();
+    $this->assertTrue($pager->valid());
+    $this->assertEquals(0, $pager->key());
+  }
+
+  public function testFromHref() {
+    $url = '/mocks';
+    $pager = new Mock_Pager($url, $this->client);
+    $pager->_loadFrom($url);
+
+    $this->assertEquals($url, $pager->getHref());
+    $this->assertEquals(6, $pager->count(), 'Returns correct count');
+    $this->assertIteratesCorrectly($pager, 6);
+  }
+
+  public function testFromStub() {
+    $url = '/mocks';
+    $stub = new Recurly_Stub('mocks', $url, $this->client);
+    $pager = $stub->get();
+    $this->assertInstanceOf('Mock_Pager', $pager);
+
+    $this->assertEquals($url, $pager->getHref());
+    // TODO: It seems to load the first page of the results, but doesn't have a
+    // link to the next page. These tests document the desired behavior but are
+    // currently broken:
+    $this->assertEquals($pager->count(), 6, 'Returns correct count');
```

Comment:
i think i had the test commented out when i wrote that. i think it can be removed.